docs/content/guide/migration.ngdoc

```diff
@@ -30,6 +31,967 @@ which drives many of these changes.
+  if a setting is not applied on `ngModelOptions`, it will be inherited from the nearest
+  `ngModelOptions` ancestor. Previously, it would default to `undefined`.
+  The programmatic API for `ngModelOptions` has also changed.
+
+- **ngTransclude**: Due to [32aa7e](https://github.com/angular/angular.js/commit/32aa7e7395527624119e3917c54ee43b4d219301),
+  if you only provide whitespace as the transclusion content, it will be assumed to be empty and the
+  fallback content will be used instead. Previously, whitespace only transclusion would be treated as
+  the transclusion being "not empty", which meant that fallback content was not used in that case.
+
+- **$location**: Due to [aa077e8](https://github.com/angular/angular.js/commit/aa077e81129c740041438688dff2e8d20c3d7b52),
+  the default hash-prefix used for `$location` hash-bang URLs has changed from the empty string
+  (`''`) to the bang (`'!'`).
+
+- **jqLite**:
+    - Due to [fc0c11](https://github.com/angular/angular.js/commit/fc0c11db845d53061430b7f05e773dcb3fb5b860),
+      jqLite will camelCase the keys passed to the `.data()` method, in the same way as jQuery 3+
+      does; i.e. single hyphens followed by a lowercase letter will be converted to an uppercase
+      letter. Previously, keys passed to `.data()` were left untouched.
+
+    - Due to [73050c](https://github.com/angular/angular.js/commit/73050cdda04675bfa6705dc841ddbbb6919eb048),
+      the way jqLite camelCases keys passed to `.css()` is aligned with jQuery. Previously, when
+      using Angular without jQuery, `.css()` would camelCase keys more aggressively. Now, only a
+      single hyphen followed by a lowercase letter is getting transformed. This change also affects
+      other APIs that rely on the `.css()` method, such as `ngStyle`.
+
+    - Due to [3faf45](https://github.com/angular/angular.js/commit/3faf4505732758165083c9d21de71fa9b6983f4a),
+      calling `.attr(attrName, '')` (with `attrName` being a boolean attribute) will no longer
+      remove the attribute, but set it to its lowercase name as happens for every non-empty string.
+      Previously, calling `.attr(attrName, '')` would remove the boolean attribute.
+
+    - Due to [d882fd](https://github.com/angular/angular.js/commit/d882fde2e532216e7cf424495db1ccb5be1789f8),
+      calling the `.val()` getter on a jqLite element representing a `<select multiple>` element
+      with no options chosen will return an empty array. Previously, it would return `null`.
+
+- **ngAria/ngModel**: Due to [975a61](https://github.com/angular/angular.js/commit/975a6170efceb2a5e6377c57329731c0636eb8c8),
+  custom `checkbox`-shaped controls (e.g. checkboxes, menuitemcheckboxes), no longer have a custom
+  `$isEmpty()` method on their `NgModelController` that checks for `value === false`. Unless
+  overwritten, the default `$isEmpty()` method will be used, which treats `undefined`, `null`, `NaN`
+  and `''` as "empty".
+
+
+Below is the full list of breaking changes:
```

Comment:
It might be worth putting in a mini-contents list here with links to each of the sections below.

E.g.

* core directives
 * forms
 * input
 * ngBind
 * ngModel
 * ngModelOptions
 * ...
* core services
 * ...
etc